src/validation.h

```diff
@@ -962,7 +976,10 @@ class ChainstateManager
     std::optional<uint256> SnapshotBlockhash() const;
 
     //! Is there a snapshot in use and has it been fully validated?
-    bool IsSnapshotValidated() const EXCLUSIVE_LOCKS_REQUIRED(::cs_main) { return m_snapshot_validated; }
+    bool IsSnapshotValidated() const EXCLUSIVE_LOCKS_REQUIRED(::cs_main)
+    {
+        return m_snapshot_chainstate && m_ibd_chainstate && m_ibd_chainstate->m_disabled;
```

Comment:
Ah, yeah, good catch! I moved too fast when addressing Antoine's feedback and forgot that setting `m_disabled` on the IBD chainstate _does_ indicate full snapshot validation. It's a little confusing because setting `m_disabled` on the _snapshot_ chainstate indicates that validation failed, but this definition of `IsSnapshotValidated()` is actually still correct, I just screwed up the comment.

Reply:
Oh, that is a helpful clarification that m_disabled is set in different conditions on the different chainstates. It'd be useful to say in doxygen comment for `m_disabled` that it is is only set on background chainstate after it is validated and validation was successful, and only set on snapshot chainstate if it's invalid and needs to be deleted.